controllers/networkfence_controller.go

```diff
@@ -18,38 +18,135 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	conn "github.com/csi-addons/kubernetes-csi-addons/internal/connection"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/proto"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/util"
+	"github.com/csi-addons/spec/lib/go/identity"
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
 )
 
-// NetworkFenceReconciler reconciles a NetworkFence object
+// NetworkFenceReconciler reconciles a NetworkFence object.
 type NetworkFenceReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	Scheme   *runtime.Scheme
```

Comment:
Exported members of a struct should be documented. Make them internal?

Reply:
I think making them internal will cause issues as it has t be called in `main` too. Wdyt?

Reply:
Better document it then :)